### PR TITLE
Various optimizations and fixes

### PR DESCRIPTION
**Pull request by markh0338**
Broke out argparse to its own function
Added ability to use text file with list of devices to gather l3 facts
A few other fixes

### CODE REVIEW

File: get_l3_facts.py [hunk 1]
```diff
@@ -6,6 +10,71 @@
...
+        "-H",
+        help="Comma-delimited list of IPs and/or FQDNs to query",
+        type=arg_list,
+    )
+    group.add_argument(
+        "--file",
+        "-f",
+        help="Text file with list of IPs and/or FQDNs (one per line) to query",
+        type=str,
+    )
```
Comment (austind) [inline get_l3_facts.py]: Change option name to `--input` / `-i`

File: get_l3_facts.py [hunk 2]
```diff
@@ -171,4 +196,4 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    sys.exit(main(sys.argv))
```
Comment (austind) [inline get_l3_facts.py]: What does this give us?

File: get_l3_facts.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+#!/usr/bin/env python3
+
+import sys
+import os
```
Comment (austind) [inline get_l3_facts.py]: Alphabetize imports

File: get_l3_facts.py [hunk 4]
```diff
@@ -76,57 +144,9 @@ def save_csv(csv_path, output):
         writer.writerows(output)
 
 
-def main():
+def main(argv):
```
Comment (austind) [inline get_l3_facts.py]: No need to pass `argv`, `argparse` handles this.

File: get_l3_facts.py [hunk 5]
```diff
@@ -6,6 +10,71 @@
 import napalm
 
 
+def get_args(args):
```
Comment (austind) [inline get_l3_facts.py]: No need to take `args`, `argparse` handles it -- see comments below

File: get_l3_facts.py [hunk 6]
```diff
@@ -76,57 +144,9 @@ def save_csv(csv_path, output):
...
-    # even when dry-running with --help
-    if not args.username:
-        username = getpass.getuser()
-        args.username = input("Username [{}]: ".format(username)) or username
-    if not args.password:
-        args.password = getpass.getpass()
-    if not args.secret:
-        args.secret = getpass.getpass("Enable secret: ")
-
+    args = get_args(argv[1:])
```
Comment (austind) [inline get_l3_facts.py]: After removing `args` from `get_args()` and `main()`, call this as `get_args()`

**Review by austind — CHANGES_REQUESTED** [review]
Looks good overall. See corrections. Thanks for contributing!

**Comment by markh0338** [comment]
requested updates have been committed.

**Comment by austind** [comment]
Alphabetize imports still pending

**Comment by markh0338** [comment]
Missed that. Reordered and separated 3rd party from standard libraries

**Pull request merged**